src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse2.cs

```diff
@@ -1023,47 +1024,90 @@ public static class Sse2
         /// __m128i _mm_set1_epi8 (char a)
         ///   HELPER
         /// </summary>
-        public static Vector128<byte> SetAllVector128(byte value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<byte> SetAllVector128(byte value)
+        {
+            Vector128<byte> vector = Sse.StaticCast<uint, byte>(ConvertScalarToVector128UInt32(value));
+            vector = UnpackLow(vector, vector);
+            return Sse.StaticCast<uint, byte>(Shuffle(Sse.StaticCast<byte, uint>(UnpackLow(vector, vector)), 0));
```

Comment:
nit: These lines are hard to read.

We should introduce temps or indent these lines, or something. @CarolEidt or @eerhardt might be able to point to the "proper" formatting for this kind of code.

Reply:
The problem here is that temps tend to be followed by additional XMM register allocated in jit, my preference would be to avoid them and try to express it better with proper indentation.

Reply:
In the case of `byte` it is visible as multiple additional unnecessary `movaps xmm1, xmm0`

Reply:
@CarolEidt, does the register allocator not recognize and optimize away temps that are write-once, read-multiple?

Reply:
> does the register allocator not recognize and optimize away temps that are write-once, read-multiple?

I'm not sure what you're asking here - if it is read-multiple it can't be optimized away. Indeed, if it's used at all, it can't really be "optimized away". However, note that all the interim values here will require allocation, and may be spilled to temps anyway.
If it is write-once, read-once (as I think you would get from the above), it should *theoretically* behave the same in the register allocator, but won't always be due to differences in the order of the trees in the IR.
In any case, we should write the code cleanly, and file an issue (*with* a buildable test case) when there are unnecessary moves.